kernel/arch/dreamcast/kernel/irq.c

```diff
@@ -83,22 +83,71 @@ int trapa_set_handler(irq_t code, irq_handler hnd, void *data) {
     return 0;
 }
 
+/* Get a string description of the exception */
+static char *irq_exception_string(int evt) {
+    switch(evt) {
+        case EXC_ILLEGAL_INSTR:
+            return "Illegal instruction";
+        case EXC_SLOT_ILLEGAL_INSTR:
+            return "Slot illegal instruction";
+        case EXC_GENERAL_FPU:
+            return "General FPU exception";
+        case EXC_SLOT_FPU:
+            return "Slot FPU exception";
+        case EXC_DATA_ADDRESS_READ:
+            return "Data address error (read)";
+        case EXC_DATA_ADDRESS_WRITE:
+            return "Data address error (write)";
+        case EXC_DTLB_MISS_READ:
+            return "Data TLB miss (read)";
+        case EXC_DTLB_MISS_WRITE:
+            return "Data TLB miss (write)";
+        case EXC_DTLB_PV_READ:
+            return "Data TLB protection violation (read)";
+        case EXC_DTLB_PV_WRITE:
+            return "Data TLB protection violation (write)";
+        case EXC_FPU:
+            return "FPU exception";
+        case EXC_INITIAL_PAGE_WRITE:
+            return "Initial page write exception";
+        default:
```

Comment:
```suggestion
        case EXC_INITIAL_PAGE_WRITE:
            return "Initial page write exception";
        case EXC_TRAPA:
            return "Unconditional trap (trapa)";
        case EXC_USER_BREAK_PRE:
        case EXC_USER_BREAK_POST:
            return "User break";
        default:
```
Added in the irq_completion_code ones, just to round it out. Not sure if they'd be visible in the same unhandled exception flow, but I don't see why not.